### PR TITLE
Added mergerfs/unionfs option for deletion ability. 

### PR DESCRIPTION
**Pull request by Dulanic**
Set pool_choice to change the ability to use mergerfs vs unionfs. Tested and both options are working. Also moved options up in config to top to consolidate and make easier to read.

image

### CODE REVIEW

**Comment by Dulanic** [comment]
And more fully tested so this wont be an issue :) default will be /w unionfs /w options for mergerfs.

**Comment by madslundt** [comment]
That is really awesome. Thank you.

**Comment by Dulanic** [comment]
Will do

File: setup.sh [hunk 1]
```diff
@@ -49,6 +49,18 @@ if [ ! -d "${rclone_dir}" ]; then
...
 fi
 
+if [ "$(pool_choice)" = "1" ]; then
+    if [ -f "${mfs_bin}" ]; then
+        git clone https://github.com/trapexit/mergerfs.git
+        cd mergerfs
+        sudo make install-build-pkgs
+        # build-essential git g++ debhelper libattr1-dev python automake libtool lsb-release
+        make deb
+        sudo dpkg -i ../mergerfs_version_arch.deb
```
Comment (Antexa) [inline setup.sh:59]: I got several errors with this code. Why not use precompiled packages?

**Comment by Dulanic** [comment]
I can adjust I used the compile version because then don't have to worry
about varying distributions. What error did u see? I also corrected an error in my original code that was always compiling. That's in next PR.

**Comment by Antexa** [comment]
Nevermind, it seems to be correct now... I may have missed something the first time around.

File: setup.sh (hunk 1 above)
Comment (Antexa) [reply setup.sh:59]: Seems to be fixed, I must have made a mistake.

**Comment by Dulanic** [comment]
OK, if you see a problem let me know for sure. May need to ensure the dependencies exist too.... I will look into that. Until @madslundt get's a chance to look over my PR https://github.com/madslundt/cloud-media-scripts/pull/45 the script is slightly off that it tried to build the package if you do have it, not if your don't. One day I'll be better at ensuring I don't break something lol

apt-get install g++ pkg-config git git-buildpackage pandoc debhelper libfuse-dev libattr1-dev -y

**Pull request merged**